src/kuzzleDataCollection.js

```diff
@@ -447,6 +445,128 @@ KuzzleDataCollection.prototype.replaceDocument = function (documentId, content,
 };
 
 /**
+ * Executes an advanced search on the data collection.
+ *
+ * /!\ There is a small delay between documents creation and their existence in our advanced search layer,
+ * usually a couple of seconds.
+ * That means that a document that was just been created won’t be returned by this function.
+ *
+ * @param {object} filters - Filters in Elasticsearch Query DSL format
+ * @param {object} [options] - Optional parameters
+ * @param {responseCallback} cb - Handles the query response
+ */
+
+KuzzleDataCollection.prototype.search = function (filters, options, cb) {
+  var
+    query,
+    self = this;
+
+  if (!cb && typeof options === 'function') {
+    cb = options;
+    options = {};
+  }
+
+  self.kuzzle.callbackRequired('KuzzleDataCollection.search', cb);
+
+  query = self.kuzzle.addHeaders({body: filters}, this.headers);
+
+
+  self.kuzzle.query(this.buildQueryArgs('read', 'search'), query, options, function (error, result) {
+    var documents = [];
+
+    if (error) {
+      return cb(error);
+    }
+
+    result.result.hits.forEach(function (doc) {
+      var newDocument = new KuzzleDocument(self, doc._id, doc._source);
+
+      newDocument.version = doc._version;
+
+      documents.push(newDocument);
+    });
+
+    if (result.result['_scroll_id']) {
+      options.scrollId = result.result['_scroll_id'];
+    }
+
+    cb(null, new KuzzleSearchResult(
+      self,
+      result.result.total,
+      documents,
+      result.result.aggregations ? result.result.aggregations : [],
+      {options: options, filters: filters}
+    ));
+  });
+};
+
+/**
+ * Scroll into a search result
```

Comment:
Can we write a more detailed description for this method? (would be handy not to have to go to the docs)